### PR TITLE
Separation of core from proto compiler output

### PR DESCRIPTION
**Pull request by dgquintas**

### CODE REVIEW

File: include/grpc++/impl/codegen/channel_interface.h [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+template <class W>
+class ClientAsyncWriter;
+template <class W, class R>
+class ClientAsyncReaderWriter;
+template <class R>
+class ClientAsyncResponseReader;
+
+/// Codegen interface for \a grpc::Channel.
+class ChannelInterface {
+ public:
```
Comment (yang-g) [inline include/grpc++/impl/codegen/channel_interface.h]: Do you need a virtual destructor for this?

File: include/grpc++/channel.h [hunk 2]
```diff
@@ -74,61 +55,28 @@ class Channel GRPC_FINAL : public GrpcLibrary,
...
+  void PerformOpsOnCall(CallOpSetInterface* ops, Call* call) override;
+  void* RegisterMethod(const char* method) override;
 
   void NotifyOnStateChangeImpl(grpc_connectivity_state last_observed,
                                gpr_timespec deadline, CompletionQueue* cq,
-                               void* tag);
+                               void* tag) override;
   bool WaitForStateChangeImpl(grpc_connectivity_state last_observed,
-                              gpr_timespec deadline);
+                              gpr_timespec deadline) override;
```
Comment (yang-g) [inline include/grpc++/channel.h]: Use GRPC_OVERRIDE instead of override?

Comment (dgquintas) [inline include/grpc++/channel.h]: yes :) you found my secret PR! It's in semi-draft form. I used "override" out of laziness to not look for where GRPC_OVERRIDE was defined, but yes, will change before actually sending out the PR for review.

File: include/grpc++/impl/codegen/channel_interface.h (hunk 1 above)
Comment (dgquintas) [inline include/grpc++/impl/codegen/channel_interface.h]: Yes, thanks!

**Pull request merged**